iceoryx_posh/test/moduletests/test_popo_active_call_set.cpp

```diff
@@ -0,0 +1,986 @@
+{
+    template <typename EventType>
+    static void doIt(TestActiveCallSet& sut, std::vector<SimpleEventClass>& events, const EventType event)
+    {
+        EXPECT_THAT(sut.attachEvent(events[N], event, ActiveCallSet_test::triggerCallback<N>).has_error(), Eq(false));
+        AttachEvent<N - 1U>::doIt(sut, events, event);
+    }
+};
+
+template <>
+struct AttachEvent<0U>
+{
+    template <typename EventType>
+    static void doIt(TestActiveCallSet& sut, std::vector<SimpleEventClass>& events, const EventType event)
+    {
+        EXPECT_THAT(sut.attachEvent(events[0U], event, ActiveCallSet_test::triggerCallback<0U>).has_error(), Eq(false));
+    }
+};
+} // namespace
+
+
+//////////////////////////////////
+// BEGIN attach / detach
+//////////////////////////////////
+TEST_F(ActiveCallSet_test, CapacityIsEqualToMAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET)
+{
+    EXPECT_THAT(m_sut->capacity(), Eq(iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET));
+}
+
+TEST_F(ActiveCallSet_test, IsEmptyWhenConstructed)
+{
+    EXPECT_THAT(m_sut->size(), Eq(0U));
+}
+
+TEST_F(ActiveCallSet_test, AttachingWithoutEnumIfEnoughSpaceAvailableWorks)
+{
+    EXPECT_FALSE(m_sut->attachEvent(m_simpleEvents[0U], ActiveCallSet_test::triggerCallback<0U>).has_error());
+    EXPECT_THAT(m_sut->size(), Eq(1U));
+}
+
+TEST_F(ActiveCallSet_test, AttachWithoutEnumTillCapacityIsFullWorks)
+{
+    fillUpWithSimpleEvents();
+    EXPECT_THAT(m_sut->size(), Eq(m_sut->capacity()));
+}
+
+TEST_F(ActiveCallSet_test, DetachDecreasesSize)
+{
+    fillUpWithSimpleEvents();
+    m_sut->detachEvent(m_simpleEvents[0U]);
+    EXPECT_THAT(m_sut->size(), Eq(m_sut->capacity() - 1U));
+}
+
+TEST_F(ActiveCallSet_test, AttachWithoutEnumOneMoreThanCapacityFails)
+{
+    fillUpWithSimpleEvents();
+    auto result = m_sut->attachEvent(m_simpleEvents[m_sut->capacity()], ActiveCallSet_test::triggerCallback<0U>);
+
+    ASSERT_TRUE(result.has_error());
+    EXPECT_THAT(result.get_error(), Eq(ActiveCallSetError::ACTIVE_CALL_SET_FULL));
+}
+
+TEST_F(ActiveCallSet_test, AttachingWithEnumIfEnoughSpaceAvailableWorks)
+{
+    EXPECT_FALSE(m_sut->attachEvent(m_simpleEvents[0U], SimpleEvent::Hypnotoad, ActiveCallSet_test::triggerCallback<0U>)
+                     .has_error());
+    EXPECT_THAT(m_sut->size(), Eq(1U));
+}
+
+TEST_F(ActiveCallSet_test, AttachWithEnumTillCapacityIsFullWorks)
+{
+    EXPECT_TRUE(fillUpWithSimpleEventsWithEnum(SimpleEvent::Hypnotoad));
+}
+
+TEST_F(ActiveCallSet_test, AttachWithEnumOneMoreThanCapacityFails)
+{
+    fillUpWithSimpleEventsWithEnum(SimpleEvent::Hypnotoad);
+    auto result = m_sut->attachEvent(
+        m_simpleEvents[m_sut->capacity()], SimpleEvent::Hypnotoad, ActiveCallSet_test::triggerCallback<0U>);
+
+    ASSERT_TRUE(result.has_error());
+    EXPECT_THAT(result.get_error(), Eq(ActiveCallSetError::ACTIVE_CALL_SET_FULL));
+}
+
+TEST_F(ActiveCallSet_test, DetachMakesSpaceForAnotherAttachWithEventEnum)
+{
+    fillUpWithSimpleEventsWithEnum(SimpleEvent::Hypnotoad);
```

Comment:
It's a bit über careful, but you could add a `EXPECT_TRUE` around the call, same for all other occurrences of `fillUpWithSimpleEvents` or change the return type to `void`.

Reply:
I would explicitly refrain from this since this is arrange state and and EXPECT_TRUE would somehow pollute this state. I already tested it and ensured that this is working